pkg/provider/app_oauth_resource.go

```diff
@@ -0,0 +1,241 @@
+			resp.State.RemoveResource(ctx)
+		} else {
+			resp.Diagnostics.AddError(
+				fmt.Sprintf("Error getting OAuth provider with ID %s", data.ID.ValueString()),
+				err.Error())
+		}
+		return
+	}
+	data.ID = types.StringValue(getResp.ID)
+	data.Name = types.StringValue(getResp.Name)
+	data.OrgID = types.StringValue(getResp.OrgID)
+	data.Type = types.StringValue(getResp.Type)
+	data.ClientID = types.StringValue(getResp.ClientID)
+	data.SecureConfigID = types.StringValue(getResp.SecureConfigID)
+	data.Status = types.StringValue(getResp.Status)
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+func (r *AppOAuthResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var plan AppOAuthResourceModel
+
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var state AppOAuthResourceModel
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if plan.ID.IsNull() {
+		resp.Diagnostics.AddError(
+			"Invalid Resource ID",
+			"Cannot update OAuth provider without a valid ID.",
+		)
+		return
+	}
+
+	request := &client.UpdateAppOAuthProviderRequest{}
+	// Only update fields that have changed
+	if plan.Name != state.Name && !plan.Name.IsNull() {
+		request.Name = plan.Name.ValueString()
+	}
+	if plan.ClientSecret != state.ClientSecret && !plan.ClientSecret.IsNull() {
+		request.ClientSecret = plan.ClientSecret.ValueString()
+	}
+
+	if request.Name == "" && request.ClientSecret == "" {
+		resp.Diagnostics.AddWarning(
+			"Invalid Update Request",
+			"At least one of 'name' or 'client_secret' must be provided for update.",
+		)
+		return
+	}
+	traceAPICall("UpdateAppOAuthProvider")
+	updateResp, err := r.provider.service.UpdateAppOAuthProvider(ctx, state.ID.ValueString(), request)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error updating OAuth provider",
+			fmt.Sprintf("Could not update OAuth provider with ID %s: %s", plan.ID.ValueString(), err),
+		)
+		return
+	}
+	plan.ID = types.StringValue(updateResp.ID)
+	plan.Name = types.StringValue(updateResp.Name)
+	plan.OrgID = types.StringValue(updateResp.OrgID)
+	plan.Type = types.StringValue(updateResp.Type)
+	plan.ClientID = types.StringValue(updateResp.ClientID)
+	plan.SecureConfigID = types.StringValue(updateResp.SecureConfigID)
+	plan.Status = types.StringValue(updateResp.Status)
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
+}
+
+func (r *AppOAuthResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var data AppOAuthResourceModel
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if data.ID.IsNull() {
+		return
+	}
+
+	traceAPICall("DeleteAppOAuthProvider")
+	err := r.provider.service.DeleteAppOAuthProvider(ctx, data.ID.ValueString())
+	if err != nil {
```

Comment:
Would this work well in case you create an OAuth provider via `pulumi up`, then deleted the poor OAuth provider manually and then run `pulumi down`. I think it should just output a warning if the OAuth provider is already deleted.

Reply:
if we delete it again, response will be 404 with message
```
{
    "message": "OAuth provider not found"
}
```
so it will fails to Diags error.... no need to extend it